src/compiler/utilities.ts

```diff
@@ -6703,21 +6713,38 @@ const enum AccessKind {
     /** Writes to a variable and uses the result as an expression. E.g.: `f(x++);`. */
     ReadWrite
 }
+function accessKindForUsageChecks(node: Node) {
+    const kind = accessKind(node);
+    const { parent } = node;
+
+    switch (parent?.kind) {
+        case SyntaxKind.PostfixUnaryExpression:
+        case SyntaxKind.PrefixUnaryExpression:
+        case SyntaxKind.BinaryExpression:
+            return kind === AccessKind.ReadWrite ? writeOrReadWrite() : kind;
+        default:
+            return kind;
+    }
+
+    function writeOrReadWrite(): AccessKind {
+        // If grandparent is not an ExpressionStatement, this is used as an expression in addition to having a side effect.
+        return parent.parent && walkUpParenthesizedExpressions(parent.parent).kind === SyntaxKind.ExpressionStatement ? AccessKind.Write : AccessKind.ReadWrite;
+    }
+}
 function accessKind(node: Node): AccessKind {
     const { parent } = node;
-    if (!parent) return AccessKind.Read;
 
-    switch (parent.kind) {
+    switch (parent?.kind) {

```

Comment:
Why would parent ever be undefined here?

Reply:
Ah, didn't realize that the old code already had an early return.

Reply:
Correct, as you saw this was inherited from the existing code that had a falsy check, even though the type suggests that it cannot be nullish.

Decided to run the tests without the optional chaining, and it seems that the tests from only this file `findAllRefsImportEqualsJsonFile.ts` are failing without it, so I guess we will keep this for now.

Reply:
> Correct, as you saw this was inherited from the existing code that had a falsy check, even though the type suggests that it cannot be nullish.

FWIW this is one case where our types lie; it's just too annoying to handle a missing parent because it only happens when the node is SourceFile.